dlt/destinations/dataset.py

```diff
@@ -178,3 +317,58 @@ def dataset(
     if dataset_type == "dbapi":
         return ReadableDBAPIDataset(destination, dataset_name, schema)
     raise NotImplementedError(f"Dataset of type {dataset_type} not implemented")
+
+
+from dlt.common.destination.reference import JobClientBase
+
+
+def create_ibis_backend(
+    destination: TDestinationReferenceArg, dataset_name: str, schema: Schema
+) -> IbisBackend:
+    from dlt.common.libs.ibis import ibis
+    import duckdb
+    from dlt.destinations.impl.duckdb.factory import DuckDbCredentials
+
+    destination = Destination.from_reference(destination)
+    client = _get_client_for_destination(destination, schema, dataset_name)
+
+    if destination.destination_type in ["dlt.destinations.postgres", "dlt.destinations.duckdb"]:
+        credentials = client.config.credentials.to_native_representation()
+        ibis = ibis.connect(credentials)
+    elif destination.destination_type == "dlt.destinations.filesystem":
+        from dlt.destinations.impl.filesystem.sql_client import (
+            FilesystemClient,
+            FilesystemSqlClient,
+        )
+
+        # we create an in memory duckdb and create all tables on there
+        duck = duckdb.connect(":memory:")
+        fs_client = cast(FilesystemClient, client)
+        creds = DuckDbCredentials(duck)
+        sql_client = FilesystemSqlClient(
+            fs_client, dataset_name=fs_client.dataset_name, credentials=creds
+        )
+
+        # NOTE: we should probably have the option for the user to only select a subset of tables here
+        with sql_client as _:
+            sql_client.create_views_for_all_tables()
+        ibis = ibis.duckdb.from_connection(duck)
+
+    else:
+        raise NotImplementedError()
+
+    # NOTE: there seems to be no standardized way to set the current dataset / schema in ibis
+    ibis.raw_sql(f"SET search_path TO {dataset_name};")
```

Comment:
why? you just do:
return ibis[dataset_name]
to select namespace.
https://duckdb.org/docs/guides/python/ibis.html

Reply:
where does it say that on this page? In any case it does not work and I also looked for a bit before and could not find any standard way to select the current database (as it is called in ibis). The call the whole thing (in the case of duckdb the file) a catalog and what we call dataset is called database there. You can list the database but not set a default one.. At least I have not found a method.

Reply:
it seems like there is no way to consistently pre-select a database, and for example setting the schema on connection in snowflake for some reason also does not work, so I am opting now to not set any default schema/database on a connection and we need to tell the user to give the database name for listing and retrieving tables when using the ibis interfaces.